src/components/ha-form/ha-form-float.ts

```diff
@@ -25,33 +25,49 @@ export class HaFormFloat extends LitElement implements HaFormElement {
 
   protected render(): TemplateResult {
     return html`
-      <paper-input
+      <mwc-textfield
         .label=${this.label}
-        .value=${this._value}
+        .value=${this.data !== undefined ? this.data : ""}
         .required=${this.schema.required}
         .autoValidate=${this.schema.required}
-        @value-changed=${this._valueChanged}
-      >
-        <span suffix slot="suffix">${this.suffix}</span>
-      </paper-input>
+        .suffix=${this.suffix}
+        .validationMessage=${this.schema.required ? "Required" : undefined}
+        @change=${this._valueChanged}
+      ></mwc-textfield>
     `;
   }
 
-  private get _value() {
-    return this.data;
-  }
-
   private _valueChanged(ev: Event) {
-    const value: number | undefined = (ev.target as PaperInputElement).value
-      ? Number((ev.target as PaperInputElement).value)
-      : undefined;
-    if (this._value === value) {
+    const source = ev.target as TextField;
+    const rawValue = source.value;
+
+    let value: number | undefined;
+
+    if (rawValue !== "") {
+      value = parseFloat(rawValue);
+    }
+
+    // Detect anything changed
+    if (this.data === value) {
+      // parseFloat will drop invalid text at the end, in that case update textfield
```

Comment:
Ah, right... But it seems that those are filtered out already? I'm not able to input anything other than numbers... (except on paste... it will do the first letter of the pasted string, seems to be a bug...)

Reply:
For integer this code will strip any decimals.

For float, the validty check in mwc-textfield doesn't handle decimals when `type="number"`. Just checks if valid integer. So it will keep saying error.